packages/installations/src/helpers/fid-changed.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getKey } from '../util/get-key';
+import { AppConfig } from '../interfaces/app-config';
+import { IdChangeCallbackFn } from '../functions';
+
+const FID_CHANGE_CALLBACKS: Map<string, Set<IdChangeCallbackFn>> = new Map();
+
+/**
+ * Calls the onIdChange callbacks with the new FID value, and broadcasts the
+ * change to other tabs.
+ */
+export function fidChanged(appConfig: AppConfig, fid: string): void {
+  const key = getKey(appConfig);
+
+  callFidChangeCallbacks(key, fid);
+  broadcastFidChange(key, fid);
+}
+
+export function addCallback(
+  appConfig: AppConfig,
+  callback: IdChangeCallbackFn
+): void {
+  // Open the broadcast channel if it's not already open,
+  // to be able to listen to change events from other tabs.
+  getBroadcastChannel();
+
+  const key = getKey(appConfig);
+
+  if (!FID_CHANGE_CALLBACKS.has(key)) {
+    FID_CHANGE_CALLBACKS.set(key, new Set());
+  }
+  FID_CHANGE_CALLBACKS.get(key)!.add(callback);
```

Comment:
You could probably avoid the double lookup in this idiom, not sure how much it actually helps but you needn't look up the thing you just inserted...

```
  let callbacks = FID_CHANGE_CALLBACKS.get(key);
  if (callbacks == null) {
    callbacks = new Set();
    FID_CHANGE_CALLBACKS.set(key, callbacks);
  }
  callbacks.add(callback);
```

Reply:
Done.